worker-test/src/test/java/com/hpe/caf/worker/workertest/AdjustRetryLimitIT.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2015-2023 Open Text.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hpe.caf.worker.workertest;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.AMQP;
+import com.rabbitmq.client.Consumer;
+import com.rabbitmq.client.ShutdownSignalException;
+import com.rabbitmq.client.Envelope;
+import org.json.JSONObject;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.concurrent.TimeoutException;
+
+public class AdjustRetryLimitIT extends TestWorkerTestBase {
+    private static final String POISON_ERROR_MESSAGE = "could not process the item.";
+    private static final String TEST_WORKER_RESULT = "TestWorkerResult";
+
+    // Test to confirm that changing the retry limit to below the current retry count will throw a poison message as expected.
+    @Test
+    public void getWorkerNameInPoisonMessageTest() throws IOException, TimeoutException {
```

Comment:
Let's split this into two tests, think about how you can avoid code duplication between the two tests.